woocommerce/woocommerce
# fix: remove help panel user meta

### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.



### Changes proposed in this Pull Request:



This PR adds a WC update routine that removes the meta key associated with the now-unused Help Panel spotlight feature.

Note that while the user meta is `help_panel_highlight_shown`, the database key is actually `woocommerce_admin_help_panel_highlight_shown` since it is automatically pre-pended.


Closes https://github.com/woocommerce/woocommerce/issues/43300.



### How to test the changes in this Pull Request:



Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

1. Start with WC 9.2.3 or earlier, since the Help Panel spotlight was removed in 9.3.0 and you won't be able to trigger the writing of that user meta if you can't access the Help Panel spotlight.
2. Trigger the spotlight by visiting the Add Product task via the Home Screen Task List, or any other task
3. Do not complete the task
4. Go back to the Home Screen, then visit the task again
5. You should see the spotlight tour
6. (Optional) check that the spotlight tour meta has been written using this in the browser console: `(await wp.data.resolveSelect('core').getCurrentUser()).woocommerce_meta`. "help_panel_highlight_shown" should contain "yes".
7. Upgrade WC to [this build](https://github.com/woocommerce/woocommerce/actions/runs/10785852873/artifacts/1912435030) 
8. Trigger the database upgrade in the notice
9. Assert that the meta has been successfully removed by checking the logs in WooCommerce Status, 'wc-updater' and also that it doesn't show up using the browser console snippet from above.

![image](https://github.com/user-attachments/assets/22b418dc-304e-4a7a-9b84-fcf91018a3d9)





### Changelog entry







<details>

<summary>Changelog Entry Details</summary>

#### Significance




#### Type




#### Message 

</details>

<details>

<summary>Changelog Entry Comment</summary>

#### Comment 

</details>
